src/newTest/java/org/opensearch/test/framework/cluster/LocalOpenSearchCluster.java

```diff
@@ -0,0 +1,610 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.test.framework.cluster;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Random;
+import java.util.SortedSet;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchTimeoutException;
+import org.opensearch.action.admin.cluster.health.ClusterHealthResponse;
+import org.opensearch.action.admin.cluster.node.info.NodeInfo;
+import org.opensearch.action.admin.cluster.node.info.NodesInfoRequest;
+import org.opensearch.action.admin.cluster.node.info.NodesInfoResponse;
+import org.opensearch.action.admin.indices.template.put.PutIndexTemplateRequest;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+import org.opensearch.client.AdminClient;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.health.ClusterHealthStatus;
+import org.opensearch.cluster.node.DiscoveryNodeRole;
+import org.opensearch.common.Strings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.transport.TransportAddress;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.http.BindHttpException;
+import org.opensearch.http.HttpInfo;
+import org.opensearch.node.PluginAwareNode;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.test.framework.certificate.TestCertificates;
+import org.opensearch.test.framework.cluster.ClusterConfiguration.NodeSettings;
+import org.opensearch.transport.BindTransportException;
+import org.opensearch.transport.TransportInfo;
+
+import com.google.common.net.InetAddresses;
+
+
+public class LocalOpenSearchCluster {
+
+    static {
+        System.setProperty("opensearch.enforce.bootstrap.checks", "true");
+    }
+
+    private static final Logger log = LogManager.getLogger(LocalOpenSearchCluster.class);
+
+    private final String clusterName;
+    private final ClusterConfiguration clusterConfiguration;
+    private final NodeSettingsSupplier nodeSettingsSupplier;
+    private final List<Class<? extends Plugin>> additionalPlugins;
+    private final List<Node> allNodes = new ArrayList<>();
+    private final List<Node> masterNodes = new ArrayList<>();
+    private final List<Node> dataNodes = new ArrayList<>();
+    private final List<Node> clientNodes = new ArrayList<>();
```

Comment:
Rather than attempting to track this information in multiple places, keep only `allNodes` and expose properties for clusterManagers or dataNodes that filter the collection based on type of the nodes

Reply:
Done